adapter/internal/operator/controllers/dp/jwtissuer_controller.go

```diff
@@ -0,0 +1,250 @@
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	jwtIssuerIndex       = "jwtIssuerIndex"
+	secretJWTIssuerIndex = "secretJWTIssuerIndex"
+	configmapIssuerIndex = "configmapIssuerIndex"
+)
+
+// JWTIssuerReconciler reconciles a JWTIssuer object
+type JWTIssuerReconciler struct {
+	client client.Client
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=dp.wso2.com,resources=jwtissuers,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=dp.wso2.com,resources=jwtissuers/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=dp.wso2.com,resources=jwtissuers/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the JWTIssuer object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.0/pkg/reconcile
+func (r *JWTIssuerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	_ = log.FromContext(ctx)
+	var err error
+
+	loggers.LoggerAPKOperator.Debugf("Reconciling jwtIssuer: %v", req.NamespacedName.String())
+
+	jwtKey := req.NamespacedName
+	var jwtIssuerList = new(dpv1alpha1.JWTIssuerList)
+	if err := r.client.List(ctx, jwtIssuerList); err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to get jwtIssuer %s/%s",
+			jwtKey.Namespace, jwtKey.Name)
+	}
+	jwtIssuerMapping := make(dpv1alpha1.JWTIssuerMapping)
+	jwtIssuerMapping, err = getJWTIssuers(ctx, r.client, jwtKey)
+	if err != nil {
+		loggers.LoggerAPKOperator.ErrorC(logging.GetErrorByCode(2606, err.Error()))
+		return ctrl.Result{}, err
+	}
+	UpdateEnforcerJWTIssuers(jwtIssuerMapping)
+	return ctrl.Result{}, nil
+}
+
+// NewJWTIssuerReconciler creates a new Application controller instance.
+func NewJWTIssuerReconciler(mgr manager.Manager) error {
+	r := &JWTIssuerReconciler{
+		client: mgr.GetClient(),
+	}
+	ctx := context.Background()
+
+	if err := addJWTIssuerIndexes(ctx, mgr); err != nil {
+		loggers.LoggerAPKOperator.ErrorC(logging.GetErrorByCode(2612, err))
+		return err
+	}
+	c, err := controller.New(constants.JWTIssuerReconciler, mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		loggers.LoggerAPKOperator.ErrorC(logging.GetErrorByCode(2606, err.Error()))
+		return err
+	}
+
+	if err := c.Watch(&source.Kind{Type: &dpv1alpha1.JWTIssuer{}}, &handler.EnqueueRequestForObject{},
+		predicate.NewPredicateFuncs(utils.FilterByNamespaces([]string{utils.GetOperatorPodNamespace()}))); err != nil {
+		loggers.LoggerAPKOperator.ErrorC(logging.GetErrorByCode(2607, err.Error()))
```

Comment:
add new error codes for new error logs. 2607 is already assigned for application error.